python/ray/data/preprocessor.py

```diff
@@ -226,59 +254,83 @@ def _transform(self, dataset: Dataset) -> Dataset:
         dataset_format = dataset._dataset_format()
         if dataset_format not in ("pandas", "arrow"):
             raise ValueError(
-                f"Unsupported Dataset format: '{dataset_format}'. Only 'pandas' and "
-                "'arrow' Dataset formats are supported."
+                f"Unsupported Dataset format: '{dataset_format}'. Only 'pandas' "
+                "and 'arrow' Dataset formats are supported."
             )
 
         transform_type = self._determine_transform_to_use(dataset_format)
 
+        # Our user facing batch format should only be pandas or numpy, other
+        # formats {arrow, simple} are internal.
         if transform_type == "pandas":
             return dataset.map_batches(self._transform_pandas, batch_format="pandas")
-        elif transform_type == "arrow":
-            return dataset.map_batches(self._transform_arrow, batch_format="pyarrow")
+        elif transform_type == "numpy":
+            return dataset.map_batches(self._transform_numpy, batch_format="numpy")
         else:
             raise ValueError(
                 "Invalid transform type returned from _determine_transform_to_use; "
-                f'"pandas" and "arrow" allowed, but got: {transform_type}'
+                f'"pandas" and "numpy" allowed, but got: {transform_type}'
             )
 
-    def _transform_batch(self, df: "DataBatchType") -> "DataBatchType":
+    def _transform_batch(self, data: "DataBatchType") -> "DataBatchType":
         import pandas as pd
 
         try:
             import pyarrow
         except ImportError:
             pyarrow = None
 
-        if isinstance(df, pd.DataFrame):
+        if isinstance(data, pd.DataFrame):
             data_format = "pandas"
-        elif pyarrow is not None and isinstance(df, pyarrow.Table):
+        elif pyarrow is not None and isinstance(data, pyarrow.Table):
             data_format = "arrow"
+        elif isinstance(data, (dict, np.ndarray)):
+            data_format = "numpy"
         else:
             raise NotImplementedError(
-                "`transform_batch` is currently only implemented for Pandas DataFrames "
-                f"and PyArrow Tables. Got {type(df)}."
+                "`transform_batch` is currently only implemented for Pandas "
+                "DataFrames, PyArrow Tables, Numpy ndarray and dictionary of "
```

Comment:
Consistent branding.
```suggestion
                "DataFrames, pyarrow Tables, NumPy ndarray and dictionary of "
```